### PR TITLE
Add scoping for location search

### PR DESCRIPTION
**Pull request by fauvel**

### CODE REVIEW

**Comment by njam** [comment]
@fvovan please review

File: tests/library/CM/Paging/Location/SearchTextTest.php [hunk 1]
```diff
@@ -50,6 +51,34 @@ public function testSearchDistance() {
...
+        $this->assertEquals(2, $source->getCount());
+
+        $source = new CM_Paging_Location_SearchText('York', CM_Model_Location::LEVEL_COUNTRY, CM_Model_Location::LEVEL_CITY);
+        $this->assertEquals(2, $source->getCount());
+
+        $source = new CM_Paging_Location_SearchText('York', CM_Model_Location::LEVEL_COUNTRY, CM_Model_Location::LEVEL_CITY, null, $country1);
+        $this->assertEquals(1, $source->getCount());
+
+        $source = new CM_Paging_Location_SearchText('York', CM_Model_Location::LEVEL_COUNTRY, CM_Model_Location::LEVEL_CITY, null, $country2);
+        $this->assertEquals(1, $source->getCount());
```
Comment (fvovan) [inline tests/library/CM/Paging/Location/SearchTextTest.php]: Maybe additionaly test if found `$city` is exactly what we are looking for (eg 'New York' or 'York') ?

**Comment by fauvel** [comment]
@fvovan please review

**Comment by fauvel** [comment]
Tagged 1.221.9

**Pull request merged**